crates/ruff_python_stdlib/src/typing.rs

```diff
@@ -198,6 +198,21 @@ pub const PEP_585_BUILTINS_ELIGIBLE: &[&[&str]] = &[
     &["typing_extensions", "Type"],
 ];
 
+// See: https://peps.python.org/pep-0585/
+pub const PEP_585_SUBSCRIPT_ELIGIBLE: &[&[&str]] = &[
+    &["typing", "DefaultDict"],
+    &["typing", "Deque"],
```

Comment:
@TylerYep - Could you speak a bit towards how this list was determined? I initially thought it'd only include builtin collections (like `List` and `Dict`), but including `DefaultDict` and `Deque` means we're asking users to import those from `collections`. So should it also include `AsyncIterator`, `Mapping`, etc. -- all the things that were moved from `typing` to `collections.abc` in Python 3.9? (I think `crates/ruff/src/rules/pyupgrade/rules/deprecated_import.rs` is our most comprehensive accounting of those moves, for reference.)

Reply:
These are all of the data structures that can be rewritten by pyupgrade into lowercase form e.g.

List[int] becomes list[int]

DefaultDict[str, bool] becomes defaultdict[str, bool]

Reply:
By that logic, should `typing.AsyncIterator` and friends also be included?

Reply:
To my knowledge, `pyupgrade` only rewrites the builtins: https://github.com/asottile/pyupgrade/blob/7c551d39dfa69851607d85392ffa296568e92436/pyupgrade/_plugins/typing_pep585.py#L15

Reply:
Ah, in that case we don't need to include Deque, DefaultDict, or FrozenSet. Fixing...